src/main/java/com/labuhn/minesweeper/application/FieldValidator.java

```diff
@@ -0,0 +1,22 @@
+package com.labuhn.minesweeper.application;
+
+import com.labuhn.minesweeper.domain.Field;
+import com.labuhn.minesweeper.domain.FlagStatus;
+
+import java.security.InvalidParameterException;
+
+public class FieldValidator {
+
+    public static void validateField(Field field){
+        if(field.getSurroundingMines() < 0 || field.getSurroundingMines() > 8){
+            throw new InvalidParameterException(String.format("Surrounding Mines must be in " +
+                    "range of 0-8. Actual fields %d", field.getSurroundingMines()));
+        }
+        if((field.getFlagStatus().equals(FlagStatus.MARKED_AS_MINE)||
```

Comment:
Just an idea. You could create a list of markedStatuses and put both marked status in there. In this if statement you could ask the list markedStatuses if it contains the field.getFlagStatus()

List<FlagStatus> markedStatuses = Arrays.asList(FlagStatus.MARKED_AS_MINE, FlagStatus.MARKED_AS_UNKNOWN)

if( !field.isCovered() && markedStatuses.contains(field.getFlagStatus() ){